### ISSUE TITLE
Odd response to unchanged data

### ISSUE THREAD

**Issue by mrDarcyMurphy** [scitran/api#103]
TL;DR: an update request is returning 404 when it maybe should be 304. 

When changing the permissions of a user on a project things work fine the first time. If I send the same request to change the permissions a second time, ie. trying to change `{access:'rw'}` to `{access:'rw'}`, then the response is `404 Element not updated...`.

Request:
- url: `http://localhost:9000/api/projects/abc123/permissions/local/foo@bar.com`
- body: `{"access":"rw"}`

1st Response:
- `200`
- `{"modified": 1}`

2nd Response:
- `404`
- `{"code":404,"uid":"foo@bar.com","detail":"Element not updated in list permissions of container projects abc123"}`

**Comment by kofalt** [comment]
Hmm. I would agree that 404 isn't appropriate in this case.

Philosophically, I would support making calls idempotent whenever possible - "ensure this state exists", contrasting with "make this state change and fail if it's already made."

Taking a [quick look](https://github.com/scitran/api/search?utf8=%E2%9C%93&q=Element+not+updated), my guess is that error code returns are appropriate for most of what this logic is rigged for, but the condition is too broad - if the database transaction failed because that state already exists, it should just 200 out.

@gsfr, thoughts?

**Comment by gsfr** [comment]
404 definitely seems odd. I agree with the desire to be idempotent. 304 would be fine, but not required.

@rentzso?

**Comment by rentzso** [comment]
yeah 404 is weird. 200 seems more reasonable for that request.

There was a subtle reason for this though.
mongo updates will return {modified: 0} every time an update doesn't modify any element and, sometimes, this is an error.

For example if we are updating a permission for a user (let's say you are changing the id) and another user with the new id already exists, we are blocking the update because we don't to override the 

I thought that, in the case mentioned in this issue, an error is weird but it would be more weird to reply to  failed updates with 200.

I don't know how to solve properly this duality. Please let me know if you have suggestions.

I was thinking now that maybe a second request after the {modified:0} result could be good to verify that the db state is consistent with the update...

**Comment by rentzso** [comment]
The above issues can be avoided if we don't allow updates on index fields.
This is what I plan to do
